### PR TITLE
Initial impl

### PR DESCRIPTION
**Pull request by peterschubert**

### CODE REVIEW

File: splunk_service.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+type splunkTransactionEvent struct {
+	splunkBaseEvent
+	transactionEvent
+}
+
+func (service *splunkService) GetTransactions(query monitoringQuery) ([]transactionEvent, error) {
+
+	envRegex := regexp.MustCompile("-u[ks]]$")
```
Comment (danielcopaciu) [inline splunk_service.go]: Can this be a global variable?

File: splunk_service.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+				transactions = append(transactions, transaction)
+			}
+		}
+	}
+	return transactions, nil
+}
+
+func (service *splunkService) GetLastEvent(query monitoringQuery) (*publishEvent, error) {
+
+	envRegex := regexp.MustCompile("-u[ks]]$")
```
Comment (danielcopaciu) [inline splunk_service.go]: Same here

Comment (peterschubert) [reply splunk_service.go]: done

File: handler.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	query := monitoringQuery{ContentType: contentType, UUIDs: uuids}
+	if interval != "" {
+		query.EarliestTime = "-" + interval
+	}
+	transactions, err := handler.splunkService.GetTransactions(query)
+
+	switch err {
+	case nil:
+		writer.WriteHeader(http.StatusOK)
+		msg, _ := json.Marshal(transactions)
```
[the remark this answers is not in the document]
Comment (peterschubert) [reply handler.go]: done

File: handler.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	query := monitoringQuery{ContentType: contentType}
+	if interval != "" {
+		query.EarliestTime = "-" + interval
+	}
+	publishEvent, err := handler.splunkService.GetLastEvent(query)
+
+	switch err {
+	case nil:
+		writer.WriteHeader(http.StatusOK)
+		msg, _ := json.Marshal(*publishEvent)
```
[the remark this answers is not in the document]
Comment (peterschubert) [reply handler.go]: done

File: splunk_service.go (hunk 1 above)
Comment (peterschubert) [reply splunk_service.go]: done

File: main.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		EnvVar: "SPLUNK_USER",
+	})
+	splunkPassword := app.String(cli.StringOpt{
+		Name:   "splunk-password",
+		Desc:   "Splunk password",
+		EnvVar: "SPLUNK_PASSWORD",
+	})
+	splunkURL := app.String(cli.StringOpt{
+		Name:   "splunk-url",
+		Desc:   "Splunk URL",
```
Comment (tamas-molnar) [inline main.go]: Maybe describe this a bit more in details - like: Splunk Rest API url?

Comment (peterschubert) [reply main.go]: done

File: README.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+`/{contentType}/transactions?[interval={relativeTime}][&uuid={uuid}]`
+
+Returns a set of unclosed transactions in a given interval
+* contentType - type of content processed in the transactions to be returned. Currently only `annotations` are supported.
+* relativeTime - earliest time to search from, in minutes or seconds. Default is `10m`
+* uuid - filter transactions by uuid; supports multiple values
+
+Response example:
+```
+[{
```
Comment (BerniVarga) [inline README.md]: Can you please update the example? (eventcount, publish_start)

File: README.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+## Other information
+
+Endpoints on this service should be used in moderation, as there are both user level and system wide limits to concurrent searches.
+As Splunk requests may fail due to these (or other) limitation, a retry mechanism is in place that attempts each query up to 3 times in a row. 
+
+### Logging
+
+* The application uses [logrus](https://github.com/Sirupsen/logrus); the log file is initialised in [main.go](main.go).
+* Logging requires an `env` app parameter, for all environments other than `local` logs are written to file.
```
Comment (BerniVarga) [inline README.md]: Does this still apply?

Comment (peterschubert) [reply README.md]: removed

File: README.md (hunk 6 above)
Comment (peterschubert) [reply README.md]: done

**Pull request merged**